### PR TITLE
Document AWS Secrets Manager Secret Store

### PR DESCRIPTION
**Pull request by sgrebnov**
Add AWS Secrets Manager Secret Store documentation

### CODE REVIEW

File: spiceaidocs/docs/secret-stores/aws-secrets-manager/index.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+```yaml
+secrets:
+  store: aws_secrets_manager
+```
+The store reads secrets named as `spice_secret_<secret-name>`, for example `dremio` login and password must be defined as `spice_secret_dremio` secret in [AWS Secrets Manager](https://console.aws.amazon.com/secretsmanager/listsecrets)
+
+<img src="/img/secrets-aws-secrets-manager-dremio-1.png" alt="" width="800" />
+
+<img src="/img/secrets-aws-secrets-manager-dremio-2.png" alt="" width="800" />
+
```
Comment (lukekim) [inline spiceaidocs/docs/secret-stores/aws-secrets-manager/index.md:22]: Add a complete example of a dataset that uses secrets manager.

Comment (sgrebnov) [reply spiceaidocs/docs/secret-stores/aws-secrets-manager/index.md:22]: Added - thank you @lukekim 

image

Also expanded first definition to provide more clarity on where `aws_secrets_manager` should be specified.
image

File: spiceaidocs/docs/secret-stores/aws-secrets-manager/index.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+Complete spicepod definition with a dataset that uses a secret from AWS Secrets Manager created above.
```
Comment (digadeesh) [inline spiceaidocs/docs/secret-stores/aws-secrets-manager/index.md]:
```suggestion
A complete spicepod definition with a dataset that uses a secret from AWS Secrets Manager created above.
```

File: spiceaidocs/docs/secret-stores/aws-secrets-manager/index.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  description: dremio taxi trips
+  params:
+    endpoint: grpc://20.163.171.81:32010
+```
+
+## AWS Secrets Manager Access
+
+To use AWS Secrets Manager, an AWS account and a user in IAM Identity Center with the `secretsmanager:GetSecretValue` permission are required. Read [Authentication and access control for AWS Secrets Manager](https://docs.aws.amazon.com/secretsmanager/latest/userguide/auth-and-access.html) for details.
+
+Use [AWS CLI](https://docs.aws.amazon.com/cli/v1/userguide/cli-chap-authentication.html) to configure AWS access:
```
Comment (digadeesh) [inline spiceaidocs/docs/secret-stores/aws-secrets-manager/index.md]:
```suggestion
Use the [AWS CLI](https://docs.aws.amazon.com/cli/v1/userguide/cli-chap-authentication.html) to configure AWS access:
```

**Review by digadeesh — DISMISSED** [review]

**Review by digadeesh — APPROVED** [review]

**Pull request merged**